assets/js/googlesitekit/modules/create-module-store.js

```diff
@@ -66,37 +66,21 @@ export const createModuleStore = ( slug, {
 		storeName,
 	} );
 
-	const STORE_NAME = [ notificationsStore.STORE_NAME ];
-	const INITIAL_STATE = [ notificationsStore.INITIAL_STATE ];
-
-	const actions = [ notificationsStore.actions ];
-	const controls = [ notificationsStore.controls ];
-	const reducer = [ notificationsStore.reducer ];
-	const resolvers = [ notificationsStore.resolvers ];
-	const selectors = [ notificationsStore.selectors ];
-
-	if ( 'undefined' !== typeof settingSlugs ) {
-		const settingsStore = createSettingsStore( 'modules', slug, 'settings', {
+	const settingsStore = ( 'undefined' !== typeof settingSlugs )
+		? createSettingsStore( 'modules', slug, 'settings', {
 			storeName,
 			settingSlugs,
-		} );
+		} )
+		: {};
+
+	const combinedStore = Data.combineStores(
+		notificationsStore,
+		settingsStore,
+	);
+
+	const STORE_NAME = settingsStore.STORE_NAME ? [ notificationsStore.STORE_NAME ] : [ notificationsStore.STORE_NAME, settingsStore.STORE_NAME ];
 
-		STORE_NAME.push( settingsStore.STORE_NAME );
-		INITIAL_STATE.push( settingsStore.INITIAL_STATE );
-		actions.push( settingsStore.actions );
-		controls.push( settingsStore.controls );
-		reducer.push( settingsStore.reducer );
-		resolvers.push( settingsStore.resolvers );
-		selectors.push( settingsStore.selectors );
-	}
+	combinedStore.STORE_NAME = Data.collectName( ...STORE_NAME );
```

Comment:
We could potentially get rid of this `collectName` logic by setting the `storeName` default ourself on top of this function (e.g. in line 64 do `storeName = storeName || `modules/${ slug }`;`). This will ensure we pass that value to `createNotificationsStore` and `createSettingsStore` and it will be returned just like that, so we already know it's going to be the same name.

Reply:
Yeah that makes sense. I've updated to simplify the storeName logic.